native/src/main/java/io/ballerina/lib/ldap/Utils.java

```diff
@@ -81,18 +82,16 @@ public static BError createError(String message, Throwable throwable) {
 
     private static BMap<BString, Object> getErrorDetails(Throwable throwable) {
         if (throwable instanceof LDAPException) {
-            String resultCode = ((LDAPException) throwable).getResultCode().getName().toUpperCase();
-            String message = throwable.getMessage();
-            return ValueCreator.createRecordValue(getModule(), ERROR_DETAILS,
-                                                  Map.of(RESULT_STATUS, resultCode, ERROR_MESSAGE, message));
+            String resultCode = ((LDAPException) throwable).getResultCode().getName().toUpperCase(Locale.ROOT);
```

Comment:
Here you can directly do something like this [1] rather than manually casting;
```java
if (throwable instanceof LDAPException ldapExp) {
    // your logic here
}
```

[1] - https://www.baeldung.com/java-pattern-matching-instanceof